src/swap/defi/0x/0xGasless.ts

```diff
@@ -71,36 +71,34 @@ export const make0xGaslessPlugin: EdgeCorePluginFactory = opts => {
         tokenId: swapRequest.fromTokenId
       })
 
-      // Amount request parameter/field name to use in the quote request
-      const amountField =
-        swapRequest.quoteFor === 'to' ? 'buyAmount' : 'sellAmount'
-
       // Get quote from ZeroXApi
       const chainId = api.getChainIdFromPluginId(
         swapRequest.fromWallet.currencyInfo.pluginId
       )
+
+      // Convert fee percentage to basis points (e.g., 0.01 -> 100 bps for 1%)
+      const swapFeeBps = Math.round(initOptions.feePercentage * 10000)
+
       const apiSwapQuote = await api.gaslessSwapQuote(chainId, {
-        [amountField]: swapNativeAmount,
-        acceptedTypes: 'metatransaction_v2',
+        sellAmount: swapNativeAmount, // v2 only supports sellAmount
         buyToken: toTokenAddress ?? NATIVE_TOKEN_ADDRESS,
         checkApproval: true,
-        feeRecipient: initOptions.feeReceiveAddress,
-        feeSellTokenPercentage: initOptions.feePercentage,
-        feeType: 'volume',
         sellToken: fromTokenAddress ?? NATIVE_TOKEN_ADDRESS,
-        takerAddress: fromWalletAddress
+        taker: fromWalletAddress,
+        swapFeeBps,
+        swapFeeRecipient: initOptions.feeReceiveAddress,
+        swapFeeToken: fromTokenAddress ?? NATIVE_TOKEN_ADDRESS,
+        tradeSurplusRecipient: initOptions.feeReceiveAddress
       })
 
-      if (!apiSwapQuote.liquidityAvailable)
+      if (!apiSwapQuote.liquidityAvailable) {
         throw new Error('No liquidity available')
+      }
 
-      // The plugin only supports gasless swaps, so if approval is required
-      // it must be gasless.
+      // Check if approval is required but gasless approval is not available
       if (
-        // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
-        apiSwapQuote.approval != null &&
-        apiSwapQuote.approval.isRequired &&
-        !apiSwapQuote.approval.isGaslessAvailable
+        apiSwapQuote.issues?.allowance != null &&
+        apiSwapQuote.approval == null
```

Comment:
Are you sure this is right? I'm reading the migration guide and it looks like the logic is different. Could you double-check this? Maybe some comments explaining what each means specifically and a "why" of the if.